firmware/Makefile

```diff
@@ -93,7 +94,7 @@ DEBUG_LINK ?= 0
 DEBUG_LOG  ?= 0
 
 CFLAGS += -Wno-sequence-point
-CFLAGS += -Iprotob -DPB_FIELD_16BIT=1
+CFLAGS += -I../vendor/nanopb -Iprotob -DPB_FIELD_32BIT=1
```

Comment:
Why do we want this change?

Reply:
The `-I../vendor/nanopb` is because we now need `pb_common.[ch]` and I didn't want to maintain the symlinks.

Now, the `-DPB_FIELD_32BIT=1` is to use `uint32_t` rather than `uint16_t` for the `size` member of a `bytes` field. Because `sizeof(uint32_t) == sizeof(size_t)`, we get identical behaviour to before.

I added this change because seems that, when we use a `uint16_t`, it triggers some GCC 4.8 bug. I say it's a bug because the warning does not make sense for that line of code (there is no comparison) and GCC 7.2 does not show this warning. But, while I found many GCC bugs with that warning, I could not find one similar to what we are seeing here. https://travis-ci.org/saleemrashid/trezor-mcu/jobs/318312962#L919

So, I'm not sure what to do about this.

Reply:
Actually, I have realized where the bug is. We are comparing `op_return_data.size < 0x100000` when that is always true because `UINT16_MAX < 0x100000`. I will revert this change and fix this warning